### ISSUE TITLE
VB Compiler doesn't complain with single-element inline initialization without values

### ISSUE THREAD

**Issue by mterwoord** [dotnet/roslyn#8106]
The following VB.NET produces an error:

```VB
dim myArray() as string = new String(2) {"Value1", "Value2"}
```
It produces the following error:
```error BC30567: Array initializer is missing 1 elements.```

This is obvious, because we're defining a 3-element array, while only specifying 2 values.

However, the following code is compiled fine:

```VB
dim x() as string = new string(0) {}
```

This is wrong, because we're defining a 1-element array, but specify 0 values.

**Comment by cston** [comment]
The behavior matches the VB Language Specification section 11.10.2.1:

> If the collection initializer is empty (that is, one that contains curly braces but no initializer list) and the bounds of the dimensions of the array being initialized are known, the empty collection initializer represents an array instance of the specified size where all the elements have been initialized to the element type's default value.
